### PR TITLE
Fix cd

### PR DESCRIPTION
**Pull request by Urhengulas**
* `**/Dockerfile`
    * copy all crates
    * unify strcuture
* add `.dockerignore`
* `.github/workflows/cd-*.yml`
    * adapt to new `Dockerfile`s
    * also run workflows if they change themselve

### CODE REVIEW

File: book/Dockerfile [hunk 1]
```diff
@@ -1,13 +1,18 @@
...
 
 FROM alpine:latest
-RUN apk add libgcc libpq
+RUN apk --no-cache add libgcc libpq
+
 COPY --from=builder /usr/local/cargo/bin/book /usr/local/bin/book
+
 EXPOSE 8080/tcp
-CMD ["book"]
+CMD [ "book" ]
```
Comment (code-mm) [inline book/Dockerfile:18]: Shouldn't this be `ENTRYPOINT [ "book" ]`?

File: borrow/Dockerfile [hunk 2]
```diff
@@ -1,13 +1,18 @@
...
 
 FROM alpine:latest
-RUN apk add libgcc libpq
+RUN apk --no-cache add libgcc libpq
+
 COPY --from=builder /usr/local/cargo/bin/borrow /usr/local/bin/borrow
+
 EXPOSE 8080/tcp
-CMD ["borrow"]
+CMD [ "borrow" ]
```
Comment (code-mm) [inline borrow/Dockerfile:18]: Shouldn't this be `ENTRYPOINT [ "borrow" ]`?

File: identity/Dockerfile [hunk 3]
```diff
@@ -1,13 +1,18 @@
...
 
 FROM alpine:latest
-RUN apk add libgcc libpq
+RUN apk --no-cache add libgcc libpq
+
 COPY --from=builder /usr/local/cargo/bin/identity /usr/local/bin/identity
+
 EXPOSE 8080/tcp
-CMD ["identity"]
+CMD [ "identity" ]
```
Comment (code-mm) [inline identity/Dockerfile:18]: Shouldn't this be `ENTRYPOINT [ "identity" ]`?

File: notification/Dockerfile [hunk 4]
```diff
@@ -1,13 +1,18 @@
...
 
 FROM alpine:latest
-RUN apk add libgcc libpq
+RUN apk --no-cache add libgcc libpq
+
 COPY --from=builder /usr/local/cargo/bin/notification /usr/local/bin/notification
+
 EXPOSE 8080/tcp
-CMD ["notification"]
+CMD [ "notification" ]
```
Comment (code-mm) [inline notification/Dockerfile:18]: Shouldn't this be `ENTRYPOINT [ "notification" ]`?

**Review by code-mm — CHANGES_REQUESTED** [review]
From my understanding this `CMD` should be `ENTRYPOINT`. The `api` services already has it.

**Comment by code-mm** [comment]
General thing, because we didn't talk about this. I usually keep a blank line at the end of all files. What is your opinion on this?

File: book/Dockerfile (hunk 1 above)
Comment (Urhengulas) [reply book/Dockerfile:18]: True

**Comment by Urhengulas** [comment]
> General thing, because we didn't talk about this. I usually keep a blank line at the end of all files. What is your opinion on this?

Agree

**Review by code-mm — APPROVED** [review]

**Pull request merged**